chore(curriculum): add pseudo-classes and elements transcripts (#58474)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/lecture-working-with-pseudo-classes-and-pseudo-elements-in-css/672bbeb6eefd7ca9c003ea00.md

  <div>This is the only item in this container.</div>
</div>

<div class="container">
  <div>This is one of two items in this container.</div>
  <div>Here is the second item.</div>
</div>
```

Using the `:only-child` pseudo-class ensures only the `div` element with a single child is selected:

```css
.container div:only-child {
   border: 2px solid crimson;
   padding: 10px;
   background-color: lightblue;
}
```

The `:first-of-type` and `:last-of-type` pseudo-classes select the first and last occurrence of a specific element type within its parent. They are useful for applying unique styles to the first or last instance of that element type among its siblings.

In the HTML below, `:first-of-type` and `:last-of-type` applies to the first element and last element within the section element:

```html
<section>
  <h2>Introduction</h2>
  <p>This is the first paragraph in the first section</p>
  <p>This is the second paragraph in the first section</p>
</section>
<section>
  <h2>Details</h2>
  <p>This is the first paragraph in the second section.</p>
  <p>This is the second paragraph in the second section.</p>
</section>
```

Here's the CSS:

```css
section p:first-of-type {

>    background-color: lightgreen;

Reviewer: ```suggestion
  background-color: lightgreen;
```

}

section p:last-of-type {

>    background-color:lightblue;

Reviewer: ```suggestion
  background-color:lightblue;
```

}
```

> `:nth-of-type(n)` allows you to select a specific element within its parent based on its position among siblings of the same type.
> 
> For instance, in the HTML below, `:nth-of-type(2)` targets the second element in the container:

Reviewer: ```suggestion
`:nth-of-type(n)` allows you to select a specific element within its parent based on its position among siblings of the same type. For instance, in the HTML below, `:nth-of-type(2)` targets the second element in the container:
```

```html
<div class="container">

>    <p>First paragraph</p>
>    <h2>First heading</h2>
>    <p>Second paragraph</p>
>    <p>Third paragraph</p>
>    <h2>Second heading</h2>

Reviewer: ```suggestion
  <p>First paragraph</p>
  <h2>First heading</h2>
  <p>Second paragraph</p>
  <p>Third paragraph</p>
  <h2>Second heading</h2>
```

</div>
```

Here's the CSS:

```css
p:nth-of-type(2) {

>     color: red;
>     font-weight: bold;

Reviewer: ```suggestion
  color: red;
  font-weight: bold;
```

}
```

> `:only-of-type` selects an element if it’s the only one of its type within its parent. This can be useful for emphasizing single items or ensuring that they are styled differently when they’re not part of a group.

Reviewer: ```suggestion
`:only-of-type` selects an element if it's the only one of its type within its parent. This can be useful for emphasizing single items or ensuring that they are styled differently when they're not part of a group.
```

In the HTML below, there are two `div` elements with one having a single element:

```html
<div class="container">
  <p>The only paragraph</p>
</div>

<div class="container">
  <p>The first paragraph</p>
  <p>The second paragraph</p>
</div>
```

Here's the CSS that only applies to the first container:

```css
p:only-of-type {

>    border: 4px solid green;

Reviewer: ```suggestion
  border: 4px solid green;
```
